lms/djangoapps/courseware/model_data.py

```diff
@@ -77,7 +78,11 @@ def __init__(self, descriptors, course_id, user, select_for_update=False, asides
         if user.is_authenticated():
             for scope, fields in self._fields_to_cache().items():
                 for field_object in self._retrieve_fields(scope, fields):
-                    self.cache[self._cache_key_from_field_object(scope, field_object)] = field_object
+                    cache_key = self._cache_key_from_field_object(scope, field_object)
+                    self.cache[cache_key] = field_object
+                    if scope == Scope.user_state:
+                        _scope, location = cache_key
```

Comment:
Right. I originally did it this way because I incorrectly thought `field_object.module_state_key` would come back as a unicode object. Changing.